src/python/pants/backend/python/tasks2/pytest_run.py

```diff
@@ -541,7 +586,7 @@ def _do_run_tests(self, targets):
       for options in self.get_options().options + self.get_passthru_args():
         args.extend(safe_shlex_split(options))
       args.extend(test_args)
-      args.extend(sources_map.keys())
+      args.extend(sorted(sources_map.keys()))
```

Comment:
Rather than sorting files, this should probably preserve the ordering of the target list by using `OrderedDict` as the `sources_map`?